### PR TITLE
[Remove Stream/Optional usage] LPS-174418 Remove usages of Stream for module: sharing-taglib

### PR DESCRIPTION
**Pull request by W2jx**

### CODE REVIEW

**Comment by zhangyan0701** [comment]
ci:test:sharing

**Comment by catarineaguiar** [comment]
no regressions found.

**Comment by W2jx** [comment]
no conflicts(2/3 8:29)

File: modules/apps/sharing/sharing-taglib/src/main/java/com/liferay/sharing/taglib/internal/util/CollaboratorsUtil.java [hunk 1]
```diff
@@ -109,25 +104,19 @@ private static String _getPortraitURL(
...
-		).map(
-			UserLocalServiceUtil::fetchUserById
-		).filter(
-			Objects::nonNull
-		).map(
-			user -> _getUserJSONObject(user, themeDisplay)
-		).forEach(
-			jsonArray::put
-		);
+		for (SharingEntry sharingEntry :
```
Comment (dantewang) [inline modules/apps/sharing/sharing-taglib/src/main/java/com/liferay/sharing/taglib/internal/util/CollaboratorsUtil.java:109]: @W2jx You can use the following method in `JSONUtil`:

```
	public static <T> JSONArray toJSONArray(
		Collection<T> collection,
		UnsafeFunction<T, Object, Exception> unsafeFunction, Log log) {
```

**Comment by dantewang** [comment]
@W2jx Please resend, thanks.

**Pull request closed without merging**